### PR TITLE
feat(mix): support constraints and use erlang v26

### PR DESCRIPTION
**Pull request by viceice**
## Changes
- use erlang v26 by default, as we use latest elixir, which now supports it.
- allow override erlang and elixir versions via constraints

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/mix/__snapshots__/artifacts.spec.ts.snap [hunk 1]
```diff
@@ -21,7 +21,7 @@ exports[`modules/manager/mix/artifacts authenticates to private repositories 2`]
     },
   },
   {
-    "cmd": "docker run --rm --name=renovate_sidecar --label=renovate_child -v "/tmp/github/some/repo":"/tmp/github/some/repo" -v "/tmp/cache":"/tmp/cache" -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w "/tmp/github/some/repo" containerbase/sidecar bash -l -c "install-tool erlang 24.3.4.2 && install-tool elixir v1.13.4 && mix hex.organization auth renovate_test --key valid_test_token && mix deps.update private_package other_package"",
+    "cmd": "docker run --rm --name=renovate_sidecar --label=renovate_child -v "/tmp/github/some/repo":"/tmp/github/some/repo" -v "/tmp/cache":"/tmp/cache" -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w "/tmp/github/some/repo" containerbase/sidecar bash -l -c "install-tool erlang 25.0.0.0 && install-tool elixir v1.13.4 && mix hex.organization auth renovate_test --key valid_test_token && mix deps.update private_package other_package"",
```
Comment (secustor) [inline lib/modules/manager/mix/__snapshots__/artifacts.spec.ts.snap:24]: How about setting a constraint for this test to keep it on 24.x.x?

**Review by secustor — COMMENTED** [review]
else lgtm

File: lib/modules/manager/mix/__snapshots__/artifacts.spec.ts.snap (hunk 1 above)
Comment (viceice) [reply lib/modules/manager/mix/__snapshots__/artifacts.spec.ts.snap:24]: 🤷‍♂️

Comment (viceice) [reply lib/modules/manager/mix/__snapshots__/artifacts.spec.ts.snap:24]: 🤷‍♂️

**Review by zharinov — APPROVED** [review]
🤷‍♂️

**Review by rarkins — APPROVED** [review]

**Pull request merged**